### ISSUE TITLE
Case insensitive Find

### ISSUE THREAD

**Issue by aaxu** [google/ci_edit#29]
Should we make it a default that when you search for a string in the text editor, it's automatically case insensitive?

**Comment by dschuyler** [comment]
Each time I start down the path of improving Find, I keep coming up with different wild ideas for how find (and replace; oh and find/replace in files) should be presented.

Yes, a case-insensitive Find should be available -- so should a case-sensitive find. 
What is a good way to allow the user to control their choice?

- it could be a config setting. That seems heavy. I wouldn't want to have to change a config to switch a search setting.
- It could be a different key in the find window. Like pressing ctrl+f does one kind of search and ctrl+somethingElse does the other kind of search. This may paint us into a corner though, there will be too many options (with all the options I think we'll want). The key combinations would get combersome.
- The search could be presented with more UI. There could be on-screen indicators of how the find will be done.
    - This has several variations that I've been daydreaming about.

Also:
How should the current search settings be presented to the user? I'd like something on-screen while the users is in InteractiveFind that shows whether case sensitivity, regex, whole word, and other options are enabled.

wdyt?

**Comment by aaxu** [comment]
Yeah I think it's too much of a hassle to have another shortcut or a whole menu for it. We can add a few buttons in the InteractiveFind window that represent the options. We'd be pressed for space, so maybe we could have a simple icon and I don't know if it's possible, but it'd also be nice if there would be help text if the user hovers over the button. I think that would be the cleanest way to go about this. I'm kind of basing my interpretations on Sublime Text cause I use that editor a lot. They support regex search, searching inside a selection, and some other features which I think are cool and maybe we can do in the future.

**Comment by dschuyler** [comment]
fyi, I made some updates to the outline of find design ideas I've been kicking around (in design/find).

**Comment by aaxu** [comment]
Oh yeah, I said it before, but I think something like sublime text would be convenient. Pressed buttons in the find window to show that certain features are on and possibly some help text or labels to appear when the user hovers over a button with the mouse, just so we can save room, though I still don't know if it's still possible yet.
